File: MitsubishiMotorsPartsECommerce.BO/OrderHeader.vb
Public Class OrderHeader
    Public Sub New()
        Me.OrderDetails = New List(Of OrderDetail)()
        Me.Payments = New Payment()
    End Sub

    Public Property OrderID As Integer
    Public Property CustomerID As Integer
    Public Property OrderDate As DateTime
    Public Property TotalAmount As Decimal
    Public Property OrderStatus As String

    Public Property Customer As Customer
    Public Property OrderDetails As IEnumerable(Of OrderDetail)
    Public Property Payments As Payment
End Class

File: MitsubishiMotorsPartsECommerce.Domain/OrderHeader.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace MitsubishiMotorsPartsECommerce.Domain.Models;

[Table("OrderHeader")]
public partial class OrderHeader
{
    [Key]
    [Column("OrderID")]
    public int OrderId { get; set; }

    [Column("CustomerID")]
    public int CustomerId { get; set; }

    [Column(TypeName = "datetime")]
    public DateTime OrderDate { get; set; }

    [Column(TypeName = "money")]
    public decimal TotalAmount { get; set; }

    [StringLength(20)]
    public string OrderStatus { get; set; } = null!;

    [ForeignKey("CustomerId")]
    [InverseProperty("OrderHeaders")]
    public virtual Customer Customer { get; set; } = null!;

    [InverseProperty("Order")]
    public virtual ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();

    [InverseProperty("Order")]
    public virtual ICollection<Payment> Payments { get; set; } = new List<Payment>();
}
